src/components/NameGenerator/NameGenerator.js

```diff
@@ -0,0 +1,29 @@
+import van from 'vanjs-core'
+import { adjectives, animals } from './words'
+import '../../style.css'
+import './nameGenerator.css'
+
+const { div, p, button } = van.tags
+const username = van.state('')
+
+const getRandomWordFromArray = (array) => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+const capitaliseWord = (word) => {
+  return word.charAt(0).toUpperCase() + word.slice(1)
+}
+
+const generateUsername = () => {
+  let namePart1 = getRandomWordFromArray(adjectives)
+  let namePart2 = getRandomWordFromArray(adjectives)
+  let namePart3 = getRandomWordFromArray(animals)
+  return (username.val = `${capitaliseWord(namePart1)}-${capitaliseWord(namePart2)}-${namePart3}`)
+}
```

Comment:
@Ania-chan That logic should be in a separate `lib` file and not in the component itself.
The name generator function will be passed as a param of the component.

The reasoning is that the generation will happen server side as we need to ensure the unicity of the name generated so 2 users can't have the same name.

Having that function is on file will allow both the design system page and the server api to have access to the same generator.